tests/test_pegen.py

```diff
@@ -532,6 +532,17 @@ def test_cut() -> None:
     ]
 
 
+def test_cut_early_exit() -> None:
```

Comment:
How does this test check that we are exiting early? Or is just for coverage?

Reply:
If the cut was not honored we would succeed to parse `(a)` as `( NAME )` and the syntax error would not be raised.

Reply:
Hummm, but the cut is honored even before this PR no? IIUC this is just optimizing when we add the error handling code

Reply:
Yes but at one point I messed up part of the logic and that was not working anymore. I think having a sanity check is a good thing.